plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1909,6 +1911,207 @@ read_only::get_table_rows_result read_only::get_table_rows( const read_only::get
+
+         fc::variant row_var;
+         if( p.json ) {
+          row_var = abis.binary_to_variant( p.table.to_string(), value, abi_serializer::create_yield_function( abi_serializer_max_time ), shorten_abi_errors );
+         } else {
+           row_var = fc::variant( value );
+         }
+         result.rows.emplace_back( std::move(row_var) );
+      }
+      return result;
+   }
+
+   ///////////////////////////////////////////////////////////
+   // ranged query
+   ///////////////////////////////////////////////////////////
+   bool has_lower_bound = p.lower_bound.has_value() && !p.lower_bound.value().empty();
+   bool has_upper_bound = p.upper_bound.has_value() && !p.upper_bound.value().empty();
+   EOS_ASSERT(has_lower_bound && has_upper_bound, chain::contract_table_query_exception,  "Unknown range: ${t} ${i}", ("t", p.table)("i", p.index_name));
+
+   vector<char> lb_key;
+   const string &lower_bound = *p.lower_bound;
+   lb_key.resize(prefix.size() + lower_bound.size());
+   memcpy(lb_key.data(), prefix.data(), prefix.size());
+   memcpy(lb_key.data() + prefix.size(), lower_bound.data(), lower_bound.size());
+
+   vector<char> ub_key;
+   const string &upper_bound = *p.upper_bound;
+   ub_key.resize(prefix.size() + upper_bound.size());
+   memcpy(ub_key.data(), prefix.data(), prefix.size());
+   memcpy(ub_key.data() + prefix.size(), upper_bound.data(), upper_bound.size());
+
+   // Iterate the range
+   vector<char> row_key;
+   vector<char> row_value;
+
+   auto lb_itr = kv_context.kv_it_create(p.code.to_uint64_t(), prefix.data(), prefix.size());
+   auto ub_itr = kv_context.kv_it_create(p.code.to_uint64_t(), prefix.data(), prefix.size());
+
+   uint32_t lb_key_size = 0;
+   uint32_t lb_value_size = 0;
+   uint32_t lb_key_actual_size = 0;
+   uint32_t ub_key_size = 0;
+   uint32_t ub_value_size = 0;
+   uint32_t ub_key_actual_size = 0;
+
+   // Find lower bound iterator
+   auto status_lb = lb_itr->kv_it_lower_bound(lb_key.data(), lb_key.size(), &lb_key_size, &lb_value_size);
+   EOS_ASSERT(status_lb != chain::kv_it_stat::iterator_erased, chain::contract_table_query_exception,  "Invalid lower bound iterator in ${t} ${i}", ("t", p.table)("i", p.index_name));
+   // Find upper bound iterator
+   auto status_ub = ub_itr->kv_it_lower_bound(ub_key.data(), ub_key.size(), &ub_key_size, &ub_value_size);
+   EOS_ASSERT(status_ub != chain::kv_it_stat::iterator_erased, chain::contract_table_query_exception,  "Invalid upper bound iterator in ${t} ${i}", ("t", p.table)("i", p.index_name));
+
+   lb_key.resize(lb_key_size);
+   status_lb = lb_itr->kv_it_key(0, lb_key.data(), lb_key_size, lb_key_actual_size);
+   ub_key.resize(ub_key_size);
+   status_ub = ub_itr->kv_it_key(0, ub_key.data(), ub_key_size, ub_key_actual_size);
+
+   // iterate the range
+   auto walk_table_row_range = [&]( auto &itr, auto &end_itr, bool reverse ) {
+      auto cur_time = fc::time_point::now();
+      auto end_time = cur_time + fc::microseconds(1000 * 1000);
```

Comment:
We have talked about future http responses to be limited by `http-max-response-time-ms` instead of hard-coded. Other hard-coded values are 10ms not 1s.